test/boost/sstable_test.hh

```diff
@@ -32,18 +32,16 @@ class column_family_test {
 public:
     column_family_test(lw_shared_ptr<replica::column_family> cf) : _cf(cf) {}
 
+    // NOTE: must run in a thread.
     void add_sstable(sstables::shared_sstable sstable) {
-        _cf->_main_sstables->insert(std::move(sstable));
-        _cf->refresh_compound_sstable_set();
+        auto new_sstables = { sstable };
+        _cf->as_table_state().on_compaction_completion(sstables::compaction_completion_desc{ .new_sstables = new_sstables }, sstables::offstrategy::no).get();
```

Comment:
wouldn't it be simpler to return a `future<>` here and let the caller `.get()` it if it's in a thread?